### PR TITLE
Fix craftInvokerFunction on JSPI (ASYNCIFY=2)

### PR DESCRIPTION
**Pull request by mboc-qt**
When the function wrapped by craftInvokerFunction doesn't return, rv is undeclared, which results in an exception being thrown at crafted function runtime.

### CODE REVIEW

File: src/embind/embind.js [hunk 1]
```diff
@@ -1131,7 +1131,7 @@ var LibraryEmbind = {
     invokerFnBody += "return Asyncify.currData ? Asyncify.whenDone().then(onDone) : onDone(" + (returns ? "rv" : "") +");\n"
 #elif ASYNCIFY == 2
     invokerFnBody += "}\n";
-    invokerFnBody += "return " + (isAsync ? "rv.then(onDone)" : "onDone(rv)") + ";";
+    invokerFnBody += `return ${isAsync ? "rv.then(onDone)" : `onDone(${returns ? "rv" : ""})`};`;
```
Comment (sbc100) [inline src/embind/embind.js:1134]: Sadly we haven't starting allowing JS template literals in our generated code yet.   We allow certain ES6 features such as arrow functions that can easily be lowered away by closure compiler, but we don't yet allow this one.

Looking at  https://caniuse.com/template-literals, it seems like perhaps we should allow it though!  

In any case, we should probably not introduce it here and now, but in a more principled fashion.

Comment (mboc-qt) [reply src/embind/embind.js:1134]: Sure thing, will use the good old string concatenation.

Comment (mboc-qt) [reply src/embind/embind.js:1134]: Also, would be happy to help with introducing that.

**Comment by mboc-qt** [comment]
Fail in test-browser-chrome looks random, am I right?

**Comment by sbc100** [comment]
> Fail in test-browser-chrome looks random, am I right?

Yup, I restarted the tests

File: src/embind/embind.js (hunk 1 above)
Comment (sbc100) [reply src/embind/embind.js]: I took a stab at it yesterday: https://github.com/emscripten-core/emscripten/pull/19285

**Comment by mboc-qt** [comment]
Shall we merge? :)

**Comment by sbc100** [comment]
Perhaps we should add a test for this case.  WDYT @brendandahl ?

**Comment by brendandahl** [comment]
Yeah, looks good. Test would be ideal.

**Comment by mboc-qt** [comment]
test_embind_jspi modified to also check this situation

**Comment by mikolaj-boc** [comment]
I think the tests flaked, could you please restart? Thanks!

**Comment by kripken** [comment]
Tests passed. PTAL @brendandahl

File: test/embind/embind_jspi_test.cpp [hunk 2]
```diff
@@ -39,9 +39,27 @@ int stdFunction(const MyClass& target, int i) {
   return i + 1;
 }
 
+EM_ASYNC_JS(void, jsSuspend, (), {
```
Comment (brendandahl) [inline test/embind/embind_jspi_test.cpp:42]: Nit: please use 2 space indent like the rest of the file.

**Review by brendandahl — APPROVED** [review]
LGTM, minor nits.

**Pull request merged**